src/Backend/PowsyblBackend.py

```diff
@@ -0,0 +1,1456 @@
+        except Exception as exc_:
+            raise BackendError(
+                f'Impossible to load the powergrid located at "{full_path}". Please '
+                f"check the file exist and that the file represent a valid pypowsybl "
+                f"grid. For your information, the error is:\n{exc_}"
+            )
+        # """
+        # Because sometimes we got negative pmin coming from matpower translation
+        # """
+
+        ind = self._grid.get_generators(all_attributes=True).index[self._grid.get_generators(all_attributes=True)['min_p'].values < 0]
+        corresp = [0 for elem in range(len(ind))]
+        self._grid.update_generators(id=ind, min_p=corresp)
+        # """
+        # We want here to change the network
+        # """
+        # with warnings.catch_warnings():
+        #     warnings.filterwarnings("ignore")
+        #     try:
+        #         ppow.loadflow.run_ac(
+        #             self._grid,
+        #             distributed_slack=self._dist_slack
+        #         )
+        #     except ppow.powerflow.LoadflowNotConverged:
+        #         ppow.loadflow.run_dc(
+        #             self._grid,
+        #             distributed_slack=self._dist_slack,
+        #         )
+
+        self.__nb_bus_before = self._grid.get_buses().shape[0]
+        self.__nb_powerline = copy.deepcopy(
+            self._grid.get_lines(all_attributes=True)[self._grid.get_lines(all_attributes=True)["r"] != 0].shape[0])
+        self._init_bus_load = self._grid.get_loads(all_attributes=True)["bus_breaker_bus_id"].values
+        self._init_bus_gen = self._grid.get_generators(all_attributes=True)["bus_breaker_bus_id"].values
+        self._init_bus_lor = self._grid.get_lines(all_attributes=True)["bus_breaker_bus1_id"].values
+        self._init_bus_lex = self._grid.get_lines(all_attributes=True)["bus_breaker_bus2_id"].values
+
+        t_for = self._grid.get_2_windings_transformers(all_attributes=True)["bus_breaker_bus1_id"].values
+        t_fex = self._grid.get_2_windings_transformers(all_attributes=True)["bus_breaker_bus2_id"].values
+        self._init_bus_lor = np.concatenate((self._init_bus_lor, t_for))
+        self._init_bus_lex = np.concatenate((self._init_bus_lex, t_fex))
+
+        # and now initialize the attributes (see list bellow)
+        if self._grid.get_3_windings_transformers().shape[0] > 0:
+            raise BackendError(f"3 windings transformers are currently not supported. "
+                               f"{self._grid.get_3_windings_transformers().shape[0]} found")
+
+        self.n_line = copy.deepcopy(self._grid.get_lines().shape[0]) + \
+                      copy.deepcopy(self._grid.get_2_windings_transformers().shape[0])
+
+        df_lines, df_transfo = self._return_real_lines_transfo()
+        self.name_line = np.array(
+            df_lines["name"].index.to_list() +
+            df_transfo["name"].index.to_list()
+        )
```

Comment:
sometimes the names of the elements are not in the json, in this case they need to be "invented", see the pandapower code for that : https://github.com/rte-france/Grid2Op/blob/0f927b10ef86ebc3d63558d7f0c9081d4e4aa751/grid2op/Backend/pandaPowerBackend.py#L481 (and following)

I will try to make a function to abstract this instead of copy pasting but for now I think copy pasting is the only solution :-/